From c1d8cc3f2fe35d506970b4f9546f125039134f16 Mon Sep 17 00:00:00 2001
From: creamlike1024 <creamlike1024@outlook.com>
Date: Sun, 14 Dec 2025 19:39:58 +0800
Subject: [PATCH] auto extend note content

---
 .../src/components/MkNoteDetailed.vue         | 23 ++++++++++++-------
 1 file changed, 15 insertions(+), 8 deletions(-)

diff --git a/packages/frontend/src/components/MkNoteDetailed.vue b/packages/frontend/src/components/MkNoteDetailed.vue
index 47bf365..c7533b2 100644
--- a/packages/frontend/src/components/MkNoteDetailed.vue
+++ b/packages/frontend/src/components/MkNoteDetailed.vue
@@ -12,9 +12,9 @@ SPDX-License-Identifier: AGPL-3.0-only
 	tabindex="0"
 >
 	<div v-if="appearNote.reply && appearNote.reply.replyId">
-		<div v-if="!conversationLoaded" style="padding: 16px">
+		<!-- <div v-if="!conversationLoaded" style="padding: 16px">
 			<MkButton style="margin: 0 auto;" primary rounded @click="loadConversation">{{ i18n.ts.loadConversation }}</MkButton>
-		</div>
+		</div> -->
 		<MkNoteSub v-for="note in conversation" :key="note.id" :class="$style.replyToMore" :note="note"/>
 	</div>
 	<MkNoteSub v-if="appearNote.replyId" :note="appearNote.reply" :class="$style.replyTo"/>
@@ -185,9 +185,9 @@ SPDX-License-Identifier: AGPL-3.0-only
 		</div>
 		<div>
 			<div v-if="tab === 'replies'">
-				<div v-if="!repliesLoaded" style="padding: 16px">
+				<!-- <div v-if="!repliesLoaded" style="padding: 16px">
 					<MkButton style="margin: 0 auto;" primary rounded @click="loadReplies">{{ i18n.ts.loadReplies }}</MkButton>
-				</div>
+				</div> -->
 				<MkNoteSub v-for="note in replies" :key="note.id" :note="note" :class="$style.reply" :detail="true"/>
 			</div>
 			<div v-else-if="tab === 'renotes'" :class="$style.tab_renotes">
@@ -594,10 +594,10 @@ function blur() {
 	rootEl.value?.blur();
 }
 
-const repliesLoaded = ref(false);
+// const repliesLoaded = ref(false);
 
 function loadReplies() {
-	repliesLoaded.value = true;
+	// repliesLoaded.value = true;
 	misskeyApi('notes/children', {
 		noteId: appearNote.id,
 		limit: 30,
@@ -606,10 +606,10 @@ function loadReplies() {
 	});
 }
 
-const conversationLoaded = ref(false);
+// const conversationLoaded = ref(false);
 
 function loadConversation() {
-	conversationLoaded.value = true;
+	// conversationLoaded.value = true;
 	if (appearNote.replyId == null) return;
 	misskeyApi('notes/conversation', {
 		noteId: appearNote.replyId,
@@ -617,6 +617,13 @@ function loadConversation() {
 		conversation.value = res.reverse();
 	});
 }
+
+// Extend note content automatically (no manual click)
+onMounted(() => {
+	loadReplies();
+	loadConversation();
+});
+
 </script>
 
 <style lang="scss" module>
-- 
2.50.1 (Apple Git-155)

